composer.json

```diff
@@ -23,7 +23,7 @@
     "require": {
         "php": ">=5.5",
         "ext-json": "*",
-        "symfony/polyfill-intl-idn": "^1.11",
+        "symfony/polyfill-intl-idn": "1.17.0",
```

Comment:
Sorry to chime in after merge: locking to a specific version of the polyfill effectively prevents installing any new(/old) versions ppl might want to use in their app. This is taking control away from users.

I'd recommend using `^1.17`.

I suppose this was done as a measure against using an `@internal` class. But the "risk" this guards against is not strong enough to justify locking all users.

The bigger risk I mentioned in my review is someone using `"replace": "symfony/polyfill-intl"`.
In order to guard from this risk, I'd recommend adding a `class_exists(Idn::class)` before using the class instead.

Reply:
Have to agree with @nicolas-grekas here, locking to a specific version is disastrous in terms of the future, and other libs which might have this same dependency..

Reply:
We will allow more versions after they are released and we can confirm comparability. Just like someone would write 5.x interested of >=5, we have to bound our versions.

Reply:
> In order to guard from this risk, I'd recommend adding a class_exists(Idn::class)

That is insufficient. What if the method is renamed, etc. No point in guessing everything possible. If someone has "replaced" the polyfill then this won't matter anyway.